### PR TITLE
add deSEC DNS plugin to certbot

### LINKED ISSUE #141: Add desec.io DNS plugin support to certbot
[linuxserverurl]: https://linuxserver.io
[linuxserver.io][linuxserverurl]

------------------------------

## Desired Behavior

Add the desec DNS plugin for certbot so that SWAG users could get automatic certs while using desec.io. https://desec.readthedocs.io/en/latest/dyndns/lets-encrypt.html

## Current Behavior

There is currently no way to automatically generate certs using desec.io as a DNS nameserver.

## Alternatives Considered

Making a docker mod, but this would be in line with the other plugins included and would benefit everyone.

### PR DESCRIPTION
**Pull request by fariszr**
[linuxserverurl]: https://linuxserver.io
[linuxserver.io][linuxserverurl]

------------------------------

 - [x] I have read the [contributing](https://github.com/linuxserver/docker-swag/blob/master/.github/CONTRIBUTING.md) guideline and understand that I have made the correct modifications

------------------------------

## Description:
fix #141
add support for desec.io for DNS auth using the deSEC certbot plugin
## Benefits of this PR and context:

this PR enables the use of https://desec.io which is a fully opensource and non-profit managed DNS host for the generation of Wildcard Let's Encrypt certificates.
This PR also exposes the option to change the deSEC api host to another domain, since deSEC can be self hosted.

## How Has This Been Tested?

i tested it on debian 11 with a custom build and a token.
the certificate has been issued successfully 

## Source / References:

https://pypi.org/project/certbot-dns-desec/

### CODE REVIEW

File: root/etc/cont-init.d/50-config [hunk 1]
```diff
@@ -122,7 +122,7 @@ if ! grep -q 'PARAMETERS' "/config/nginx/dhparams.pem"; then
 fi
 
 # check to make sure DNSPLUGIN is selected if dns validation is used
-[[ "$VALIDATION" = "dns" ]] && [[ ! "$DNSPLUGIN" =~ ^(aliyun|cloudflare|cloudxns|cpanel|digitalocean|directadmin|dnsimple|dnsmadeeasy|domeneshop|gandi|gehirn|google|he|hetzner|inwx|ionos|linode|luadns|netcup|njalla|nsone|ovh|rfc2136|route53|sakuracloud|transip|vultr)$ ]] && \
+[[ "$VALIDATION" = "dns" ]] && [[ ! "$DNSPLUGIN" =~ ^(aliyun|cloudflare|cloudxns|cpanel|digitalocean|directadmin|dnsimple|dnsmadeeasy|domeneshop|gandi|gehirn|google|he|hetzner|inwx|ionos|linode|luadns|netcup|njalla|nsone|ovh|rfc2136|route53|sakuracloud|transip|vultr|desec)$ ]] && \
```
Comment (nemchik) [inline root/etc/cont-init.d/50-config:125]:
```suggestion
[[ "$VALIDATION" = "dns" ]] && [[ ! "$DNSPLUGIN" =~ ^(aliyun|cloudflare|cloudxns|cpanel|desec|digitalocean|directadmin|dnsimple|dnsmadeeasy|domeneshop|gandi|gehirn|google|he|hetzner|inwx|ionos|linode|luadns|netcup|njalla|nsone|ovh|rfc2136|route53|sakuracloud|transip|vultr)$ ]] && \
```
Sorted alphabetically.

File: root/etc/cont-init.d/50-config [hunk 2]
```diff
@@ -236,7 +236,7 @@ if [ "$VALIDATION" = "dns" ]; then
     elif [[ "$DNSPLUGIN" =~ ^(google)$ ]]; then
         if [ -n "$PROPAGATION" ];then PROPAGATIONPARAM="--dns-${DNSPLUGIN}-propagation-seconds ${PROPAGATION}"; fi
         PREFCHAL="--dns-${DNSPLUGIN} --dns-${DNSPLUGIN}-credentials /config/dns-conf/${DNSPLUGIN}.json ${PROPAGATIONPARAM}"
-    elif [[ "$DNSPLUGIN" =~ ^(aliyun|domeneshop|he|hetzner|inwx|ionos|netcup|njalla|transip|vultr)$ ]]; then
+    elif [[ "$DNSPLUGIN" =~ ^(aliyun|domeneshop|he|hetzner|inwx|ionos|netcup|njalla|transip|vultr|desec)$ ]]; then
```
Comment (nemchik) [inline root/etc/cont-init.d/50-config:239]:
```suggestion
    elif [[ "$DNSPLUGIN" =~ ^(aliyun|desec|domeneshop|he|hetzner|inwx|ionos|netcup|njalla|transip|vultr)$ ]]; then
```
Sorted alphabetically.

File: readme-vars.yml [hunk 3]
```diff
@@ -51,7 +51,7 @@ opt_param_usage_include_env: true
...
-  - { env_var: "DNSPLUGIN", env_value: "cloudflare", desc: "Required if `VALIDATION` is set to `dns`. Options are `aliyun`, `cloudflare`, `cloudxns`, `cpanel`, `digitalocean`, `directadmin`, `dnsimple`, `dnsmadeeasy`, `domeneshop`, `gandi`, `gehirn`, `google`, `he`, `hetzner`, `inwx`, `ionos`, `linode`, `luadns`, `netcup`, `njalla`, `nsone`, `ovh`, `rfc2136`, `route53`, `sakuracloud`, `transip` and `vultr`. Also need to enter the credentials into the corresponding ini (or json for some plugins) file under `/config/dns-conf`." }
+  - { env_var: "DNSPLUGIN", env_value: "cloudflare", desc: "Required if `VALIDATION` is set to `dns`. Options are `aliyun`, `cloudflare`, `cloudxns`, `cpanel`, `desec`, `digitalocean`, `directadmin`, `dnsimple`, `dnsmadeeasy`, `domeneshop`, `gandi`, `gehirn`, `google`, `he`, `hetzner`, `inwx`, `ionos`, `linode`, `luadns`, `netcup`, `njalla`, `nsone`, `ovh`, `rfc2136`, `route53`, `sakuracloud`, `transip` and `vultr`. Also need to enter the credentials into the corresponding ini (or json for some plugins) file under `/config/dns-conf`." }
```
Comment (nemchik) [inline readme-vars.yml:54]: No issue with this line, but we also need a changelog entry towards the bottom of the file https://github.com/fariszr/docker-swag/blob/0d5f7b24b83a5dafc85d4d2b66c300453bc4aae9/readme-vars.yml#L157

Also in the future, please create a separate branch for changes in your fork. Making changes to the `master` branch in your fork can complicate changes being merged.

**Review by nemchik — CHANGES_REQUESTED** [review]

File: readme-vars.yml (hunk 3 above)
Comment (fariszr) [reply readme-vars.yml:54]: fixed

**Review by nemchik — APPROVED** [review]

**Pull request merged**